### PR TITLE
[ready] torch.* doc update for Variable/Tensor merge, and other improvements

### PR DESCRIPTION
**Pull request by ssnl**
Commits:
1. Doc changes
+ Update doc to reflect changes in Variable/Tensor merge (`-> float` functions become `-> Tensor`), and new printing style
+ Remove functions in` torch/functional.py` that are already implemented with native_function. Move their doc to `_torch_docs.py` and `_tensor_docs.py`.
+ Add `set_detault_tensor_type` doc
+ Generally improve doc, including clarifying `torch.gels` (#5431 )

2. Fix `torch.split`. The method behaves differently basing on the type of 2nd arg. 
+ This commit breaks it into two functions in ATen `split` and `split_with_sizes`. The Python API stays the same with the  if-statement in `variable.py`. 
+ To override the `torch.split` functional form, this commit also moves `from .functional import *` to after ATen bindings in `torch/__init__.py`. 
+ Finally, this commit also extends `_add_doc_str` to work on Python functions so `torch.split` doc can remain in `_torch_docs.py`.

partially addresses #5571 

cc @vishwakftw hope that this doesn't conflict with what you are working on.

### CODE REVIEW

**Comment by SsnL** [comment]
changing `torch.split`

File: torch/csrc/Module.cpp [hunk 1]
```diff
@@ -445,9 +445,19 @@ PyObject *THPModule_addDocStr(PyObject *_unused, PyObject *args)
           "method '%s' already has a docstring", m->d_method->ml_name);
     }
     m->d_method->ml_doc = doc_str;
-  } else {
-    return PyErr_Format(PyExc_TypeError,
-        "don't know how to add docstring to type '%s'", Py_TYPE(obj)->tp_name);
+  } else {  // e.g. python function
```
Comment (colesbury) [inline torch/csrc/Module.cpp]: I think we should document Python functions in Python using standard docstrings

Comment (SsnL) [reply torch/csrc/Module.cpp]: Sure, I removed it. I thought maybe keeping every torch.* doc at the same file would be easier to organize.

File: torch/__init__.py [hunk 2]
```diff
@@ -127,10 +127,54 @@ def is_storage(obj):
     return type(obj) in _storage_classes
 
 
+if sys.version_info[0] == 2:
+    _base_str_class = basestring
```
Comment (fmassa) [inline torch/__init__.py]: Can't you use the functionality provided in https://github.com/pytorch/pytorch/blob/master/torch/_six.py#L29 for this?

Comment (SsnL) [reply torch/__init__.py]: Thanks for pointing out! I wasn't aware of that :)

**Comment by SsnL** [comment]
@colesbury could you take a look at this? Most of the changes are examples formatting and `-> float` => `-> Tensor`.

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -1841,6 +1843,7 @@ def callable(a, b) -> number
     args (torch.Size or int...): the desired size
 
 Example:
```
Comment (fmassa) [inline torch/_tensor_docs.py]: Do we need two colons here? `Examples::`

**Review by colesbury — APPROVED** [review]
changes lgtm assuming the tests pass

**Pull request merged**